### PR TITLE
feat(zb-db): disable rocksdb WAL by default

### LINKED ISSUE #11455: Disable Rockdb WAL by default
**Is your feature request related to a problem? Please describe.**
We already provide a configuration flag to disable rocksdb WAL. [Our recent benchmarks](https://camunda.slack.com/archives/C037RS2JHB8/p1674200750104639) has shown that disabling rocksdb WAL provides noticeable performance (process latency)  improvement. So it would make sense to disable it by default. 

As part of this issue, also crosscheck if this is safe. See https://github.com/camunda/zeebe/issues/6161#issuecomment-771722771 more details. Bonus would be if we can test and verify our understanding of its safety is correct.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

WAL is useful to crash recovery. Since after crash we always use the last checkpoint and Zeebe's logstream to rebuild the state. there is no need to keep the WAL. As discussed in https://github.com/camunda/zeebe/issues/6161#issuecomment-771722771 it seems to be safe while taking checkpoints as well. It is not to safe to disable WAL if we add more than one column families. 

Throughput with WAL disabled
image

compared to medic-cw-05
image

- [E2E test ](https://bru-3.operate.ultrawombat.com/7b2cd3f5-cfcc-4a65-ab22-0ccaa071e68e/processes/2251799961231250) successfully ran with this change. 
- Benchmark is running.

## Related issues

closes #11455

### CODE REVIEW

**Review by ChrisKujawa — APPROVED** [review]
:rocket: 

I think we need to adjust the application.yaml comments ?

**Comment by deepthidevaki** [comment]
> I think we need to adjust the application.yaml comments ?

Yes. Totally forgot about that. Thanks.

File: zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/RocksDbConfiguration.java [hunk 1]
```diff
@@ -16,7 +16,22 @@ public final class RocksDbConfiguration {
   public static final int DEFAULT_MAX_WRITE_BUFFER_NUMBER = 6;
   public static final int DEFAULT_MIN_WRITE_BUFFER_NUMBER_TO_MERGE = 3;
   public static final boolean DEFAULT_STATISTICS_ENABLED = false;
-  public static final boolean DEFAULT_WAL_DISABLED = false;
+
+  /**
+   * WARN: It is safe to disable wal as long as there is only one column family. With more than one
```
Comment (npepinpe) [inline zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/RocksDbConfiguration.java:21]: :heart:

**Review by npepinpe — APPROVED** [review]
imo 100% safe to backport :+1:

**Comment by deepthidevaki** [comment]
> imo 100% safe to backport +1

Agree. But I don't see a need to backport this. The configuration is available in 8.1 already. So it can be configured if required.

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**